modules/rostests/apitests/shell32/LocaleTests.cpp

```diff
@@ -0,0 +1,290 @@
+    { UENV_STARTUP, { userenv, 13 /* IDS_STARTUP "Start Menu\Programs\StartUp" */, 3 } },
+    { SYSS_PROGRAMFILES, { syssetup, 3600 /* IDS_PROGRAMFILES "%SystemDrive%\Program Files" */, 2 } },
+    { SYSS_COMMONFILES, { syssetup, 3601 /* IDS_COMMONFILES "Common Files" */, 1 } },
+    { MMSY_STARTMENU, { mmsys, 5851 /* IDS_STARTMENU "Start Menu" */, 1 } },
+    { EOLD_PROGRAMS, { explorer_old, 10 /* IDS_PROGRAMS "Programs" */, 1 } },
+};
+
+static PART_MATCH PartMatches[] =
+{
+    // Start Menu
+    { { SH32_PROGRAMS, 0 }, { SH32_STARTUP, 0 } },
+    { { SH32_PROGRAMS, 0 }, { SH32_STARTMENU, 0 } },
+    { { SH32_PROGRAMS, 0 }, { SH32_ADMINTOOLS, 0 } },
+    { { SH32_PROGRAMS, 0 }, { UENV_STARTMENU, 0 } },
+    { { SH32_PROGRAMS, 0 }, { UENV_PROGRAMS, 0 } },
+    { { SH32_PROGRAMS, 0 }, { UENV_STARTUP, 0 } },
+    { { SH32_PROGRAMS, 0 }, { MMSY_STARTMENU, 0 } },
+    // Programs
+    { { SH32_PROGRAMS, 1 }, { SH32_STARTUP, 1 } },
+    { { SH32_PROGRAMS, 1 }, { SH32_ADMINTOOLS, 1 } },
+    { { SH32_PROGRAMS, 1 }, { UENV_PROGRAMS, 1 } },
+    { { SH32_PROGRAMS, 1 }, { UENV_STARTUP, 1 } },
+    { { SH32_PROGRAMS, 1 }, { EOLD_PROGRAMS, 0 } },
+    // StartUp
+    { { SH32_STARTUP, 2 }, { UENV_STARTUP, 2 } },
+    // Program Files
+    { { SH32_PROGRAM_FILES, 0 }, { SH32_PROGRAM_FILES_COMMON, 0 } },
+    { { SH32_PROGRAM_FILES, 0 }, { SYSS_PROGRAMFILES, 1 } },
+    // Common Files
+    { { SH32_PROGRAM_FILES_COMMON, 1 }, { SYSS_COMMONFILES, 0 } },
+};
+
+static DWORD CountParts(_In_ LPWSTR str)
+{
+    DWORD count = 0;
+    LPWSTR ptr = str;
+
+    if (*ptr == UNICODE_NULL)
+        return 0;
+
+    while ((ptr = wcschr(ptr, L'\\')) != NULL)
+    {
+        count++;
+        ptr++;
+    }
+
+    return count + 1;
+}
+
+static LPWSTR GetPart(_In_ LPWSTR str, _In_ SIZE_T num, _Out_ SIZE_T* len)
+{
+    DWORD count = 0;
+    LPWSTR ptr = str, next;
+
+    while (count < num && (ptr = wcschr(ptr, L'\\')) != NULL)
+    {
+        count++;
+        ptr++;
+    }
+
+    next = wcschr(ptr, L'\\');
+    *len = next ? next - ptr : wcslen(ptr);
+    return ptr;
+}
+
+static BOOL CALLBACK find_locale_id_callback(
+    _In_ HMODULE hModule, _In_ LPCWSTR type, _In_ LPCWSTR name, _In_ LANGID lang, _In_ LPARAM lParam)
+{
+    langs.insert(lang);
+    return TRUE;
+}
+
+static void SetLocale(_In_ LCID lcid)
+{
+    SetThreadLocale(lcid);
+    SetThreadUILanguage(lcid);
+    curLcid = lcid;
+}
+
+static void TEST_NumParts(void)
+{
+    for (auto i = parts.begin(); i != parts.end(); i++)
+    {
+        E_MODULE m = i->second.eModule;
+
+        if (!mod[m])
+        {
+            skip("No module for test #%d\n", i->first);
+            continue;
+        }
+
+        WCHAR szBuffer[MAX_PATH];
+
+        LoadStringW(mod[m], i->second.id, szBuffer, _countof(szBuffer));
+        ok(i->second.nParts == CountParts(szBuffer), "Locale 0x%lX: Num parts mismatch #%d - expected %lu, got %lu\n",
```

Comment:
This can be replaced with `ok_eq_size`, and `ok(wcscmp` with `ok_eq_wstr`

Reply:
Decided to keep current ones for better verbosity, as it would be harder with these to track mismatches.